text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,425 @@
+
+Combining these two criteria, for a `Range` to not be *empty*, progress from `min` to `max` must be possible (no-progess is `false`), and the iterator that realizes this progress must produce *finite* values that lie within `[min, max)`.
+
+Using this test, `Range(0, 10, -1)` is *empty* and no longer *infinite*, but so is in fact `Range(0, inf, -1)` that is also considered *infinite* currently. While one can generally never produce a complete list of elements that would incrementally move from say 0 to +Inf, we can know in the case of `Range(0, inf, -1)` that the Range must be *empty* because of the incompatible sign of the step parameter relative to the signs of `min` and `max` (progress impossible).
+Another case in which a Range is now *empty* is when the bound equality `min == max` can be evaluated meaningfully, independent of whether `step` equals e.g. `0` or `+-Inf` such as for example `Range(10, 10, 0)`. In the case of equal and finite bounds, no value can exist that satisfies `[min, min)`. In case of infinite bound values, the range is also empty because one `inf` cannot be meaningfully tested for equality to another `inf` and even if it could, the second criterion would be violated in that a sequence of values would contain non-finite values.
```

Comment:
This explains that Range(Inf, Inf, 1) is empty for two reasons: the possibility of progress can't be evaluated using the *no progress* expression, and even if it could, the second criterion requires the iterator to produce finite values.

Reply:
Okay, I understand from the point of not being able to compare infinites. I was thinking about it from both a representation standpoint (inf is the same bits as inf) and from an inductive `x +1 == x +1` for any value of x up to inf.